resources/documentation/learn_ansible.rst

```diff
@@ -691,5 +691,395 @@ Now regarding to Jinja2:
+
+  TASK [debug : Get output from a shell command] **********************************************************************************************************************************************************************************************
+  changed: [management1]
+
+  TASK [debug : My message module] ************************************************************************************************************************************************************************************************************
+  ok: [management1] =>
+    msg:
+      changed: true
+      cmd: cat /tmp/valkyrie
+      delta: '0:00:00.016261'
+      end: '2020-04-17 14:46:04.490684'
+      failed: false
+      rc: 0
+      start: '2020-04-17 14:46:04.474423'
+      stderr: ''
+      stderr_lines: []
+      stdout: lenneth
+      stdout_lines:
+      - lenneth
+
+So, you can use the register to get status (changed: true, so something happens), to get return code (rc at 0 here), stderr and stdout, if it failed, etc.
+
+For example:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "{{ my_result.stdout }}"
+
+Will display "lenneth".
+
+We are going to uses register and debug module to learn generic tasks actions.
+
+Loops
+^^^^^
+
+It is possible to make modules iterate. All possibilities are available here in `loop documentation <https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html>`_ .
+
+BlueBanquise rely on two methods: *with_items* and *loop*. *with_items* is considered replaced by the Ansible team, but since *loop*, the replacement, does not cover all the abilities of *with_items*, it is still widely used.
+
+First example is with loop:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "Values: {{ item }}"
+    loop:
+       - Blue
+       - Red
+       - Green
+       - Yellow
+
+This will execute this module 4 times, with each time an element of the list gives to loop. When using loops in ansible, the variable **item** store the value of the current loop index.
+
+It is also possible to provide loop a list from the inventory, like the one we made before:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "Values: {{ item }}"
+    loop: "{{ my_ship.equipment.sidekicks }}"
+
+Loop action can accept advanced filters or patterns. Have a look in the official documentation.
+
+with_items works the same way:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "Values: {{ item }}"
+    with_items:
+       - Blue
+       - Red
+       - Green
+       - Yellow
+
+  - name: My message module
+    debug:
+      msg: "Values: {{ item }}"
+    with_items: "{{ my_ship.equipment.sidekicks }}"
+
+So why not using loop all the time?
+
+Issue is when you need to use a conditional (see below) and that condition is used to check if the variable is defined.
+with_items will execute after the when, while loop execute before, and so could fail.
```

Comment:
are you sure?

Reply:
Try it. This was one of the main issue I had when migrating to loop, when using ansible 2.7.

Reply:
removed